src/Analyzers/Core/Analyzers/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -114,6 +114,9 @@ private void AnalyzeCompilation(CompilationAnalysisContext context)
             // in their project file to enable IDE0005 on build.
 
             var compilation = context.Compilation;
+            if (compilation.Options.WarningLevel < 8)
+                return;
```

Comment:
📝 Note that this warning only gets reported if the user has `EnforceCodeStyleInBuild` enabled. For this scenario, the analyzer will come from the _SDK_ and not from Visual Studio itself, so this fix will only roll out to users once they have a .NET SDK version which contains the fix.

Reply:
> It's also the first time we've tied a non-compiler warning to a compiler warning wave level.

I do wonder about that: is there an easy way we can tie it to `AnalysisLevel` instead?

Reply:
What is AnalysisLevel?

Reply:
`AnalysisLevel` is the more general MSBuild flag that influences all built-in analyzers, and also sets `WarningLevel`. It seems to me to be the more accurate thing to use here.

Reply:
is this a blocker for landing this PR? 

(we do need a second compiler signoff)

Reply:
Sam did some investigating, and it seems like we can't easily do this change, so I'm ok with proceeding here.